app/services/form_response.rb

```diff
@@ -39,10 +39,11 @@ def merge(other)
     end
   end
 
-  def first_error_message(key = nil)
+  def first_error_message(*keys)
     return if errors.blank?
-    key ||= errors.keys.first
-    errors[key].first
+    [*keys, errors.keys.first].each do |key|
+      return errors[key].first if errors.key?(key)
+    end
```

Comment:
I updated my initial recommendation. Here's some specs which can illustrate the problem with the original recommendation implemented here:

```rb

  describe '#first_error_message' do
    context 'with no errors' do
      subject(:first_error_message) { form_response.first_error_message }
      let(:errors) { {} }

      it { expect(first_error_message).to be_nil }
    end

    context 'with errors' do
      let(:errors) { { email: ['invalid', 'too_short'], language: ['blank'] } }

      context 'without specified key' do
        subject(:first_error_message) { form_response.first_error_message }

        it 'returns the first error of the first field' do
          expect(first_error_message).to eq('invalid')
        end
      end

      context 'with specified key' do
        subject(:first_error_message) { form_response.first_error_message(:language) }

        it 'returns the first error of the specified field' do
          expect(first_error_message).to eq('blank')
        end

        context 'with key that does not exist in set of errors' do
          subject(:first_error_message) { form_response.first_error_message(:foo) }

          it 'returns nil' do
            expect(first_error_message).to be_nil
          end
        end
      end

      context 'with multiple specified keys' do
        subject(:first_error_message) { form_response.first_error_message(:foo, :language, :email) }

        it 'returns the first error of the key which exists as an error' do
          expect(first_error_message).to eq('blank')
        end
      end
    end
  end
```

Reply:
Added that change in. It looks like it's working great. Thanks!